src/darwin/Framework/CHIP/ServerEndpoint/MTRServerEndpoint_Internal.h

```diff
@@ -0,0 +1,64 @@
+/**
+ *    Copyright (c) 2024 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#import <Foundation/Foundation.h>
+#import <Matter/MTRDeviceController.h>
+#import <Matter/MTRServerEndpoint.h>
+
+typedef NS_ENUM(NSUInteger, MTRServerEndpointState) {
+    /**
+     * The Initializing state indicates an endpoint or object that hangs off an
+     * endpoint (cluster, attribute, etc) that is still being set up by the API
+     * consumer and has not been added to a controller yet.
+     */
+    MTRServerEndpointStateInitializing,
```

Comment:
Do we need to distinguish "initializing" and "defunct"? Seems like we could just have "active" and "inactive", i.e. detaching goes back to inactive.

Reply:
Done.